listings/ch02-applications/simple_vault/src/test.cairo

```diff
@@ -0,0 +1,50 @@
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use starknet::{
+        ContractAddress,
+        array::ArrayTrait,
+        traits::{Into, TryInto},
+    };
+
+    // Simulated dispatcher for the ERC20 contract.
+    struct MockERC20Dispatcher;
+
+    impl IERC20DispatcherTrait for MockERC20Dispatcher {
+        // Simulated implementation of ERC20 methods for testing.
+        // Replace these with actual logic in a real environment.
+        fn balance_of(&self, _account: ContractAddress) -> u256 {
+            // this is a simulated balance for testing
+            1000 
+        }
+
+        fn transfer(&self, _recipient: ContractAddress, _amount: u256) -> bool {
+            true 
+        }
+
+        fn transfer_from(&self, _sender: ContractAddress, _recipient: ContractAddress, _amount: u256) -> bool {
+            true
+        }
+    }
+
+    #[test]
+    fn test_deposit_withdraw() {
+        // Set up.
+        let mut token_dispatcher = MockERC20Dispatcher {};
+        let mut contract = SimpleVaultDispatcher { token: &mut token_dispatcher, total_supply: 0, balance_of: LegacyMap::new() };
```

Comment:
Here it's ok to use a Mocked ERC20, but it's better to deploy the SimpleVault contract `deploy_syscall` and  `SimpleVaultContract::TEST_CLASS_HASH`.